docker/optimism/optimism.sh

```diff
@@ -12,72 +11,102 @@ CACHE_DIR=$(obtain_cache_dir $0)
 DEPLOYMENT_DIR="${CACHE_DIR}/deployment"
 DEPLOYMENT_CONFIG_FILE="${CACHE_DIR}/optimism-local.json"
 KEYFILE="${CACHE_DIR}/${ADDRESS}.json"
+OPTIMISM="${CACHE_DIR}/optimism"
 
 ensure_keyfile_exists ${PRIVKEY} ${KEYFILE}
 echo Beamer deployer\'s keyfile: ${KEYFILE}
 
+e2e_test_op_proof(){
+    local root=$(get_root_dir)
+    local l1_rpc=$1
+    local l2_rpc=$2
+    local privkey=$4
+    local txhash=$5
+    local relayer=${root}/relayer/relayer-node18-linux-x64
+
+    echo Starting OP relayer message prover...
+    timeout 5m bash -c "until ${relayer} prove-op-message \
+                                         --l1-rpc-url '$l1_rpc' \
+                                         --l2-rpc-url '$l2_rpc' \
+                                         --wallet-private-key '$privkey' \
+                                         --l2-transaction-hash '$txhash'; \
+    do sleep 1s; done"
+}
+
+configure_repo() {
+    repo="https://github.com/ethereum-optimism/optimism.git"
+    commit_id="6a474e36aba94f15ed90f71d1920e4b1d34513ab"
+    git clone --no-checkout ${repo} ${OPTIMISM}
+    cd ${OPTIMISM}
+    git checkout ${commit_id}
+}
+
+run_on_op_launcher () {
+    command=$1
+    docker run -v /var/run/docker.sock:/var/run/docker.sock \
+        -v ${OPTIMISM}:${OPTIMISM} \
+        --network=host optimism-launcher bash -c "${command}"
+}
+
 down() {
     echo "Shutting down the end-to-end environment"
-    make -C ${OPTIMISM}/ops down
+    run_on_op_launcher "cd ${OPTIMISM} && make devnet-down && make devnet-clean"
+}
+
+build() {
+    DOCKER_BUILDKIT=1 docker build ${ROOT}/docker/optimism -t optimism-launcher
```

Comment:
Is the build image independent of the commit ID from the optimism repo?
I am mainly thinking about a case where we already have an image (`optimism-launcher`) and then update the optimism commit ID. Would a new launcher image be needed or would the existing one be fine?

Reply:
we are using specific commit for repo and if it wont change then no problem, if we need to change it then we wouldnt actually build it again as up checks for image